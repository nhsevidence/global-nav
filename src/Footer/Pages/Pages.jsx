import React, { Component } from "react";
import PropTypes from "prop-types";

import styles from "./Pages.module.scss";

import pages from "./pages.json";
import TrackedLink from "../../TrackedLink";
import { footerClickEventAction } from "../../tracker";

export default class Pages extends Component {
	render() {
		return (
			<nav className={styles.wrapper}>
				<ul className={styles.list}>
					{pages.map(
						function(link) {
							let href = link.href;
							if (link.hrefOverrides && this.props.service) {
								href = link.hrefOverrides[this.props.service] || href;
							}
							return (
								<li key={link.text}>
									<TrackedLink
										href={href}
										eventAction={footerClickEventAction}
										eventLabel={link.text}
									>
										{link.text}
									</TrackedLink>
								</li>
							);
						}.bind(this)
					)}
				</ul>
			</nav>
		);
	}
}

Pages.propTypes = {
	service: PropTypes.string
};
