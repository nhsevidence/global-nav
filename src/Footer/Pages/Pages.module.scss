.wrapper {
  margin: rem(0 0 $spacing-medium);
}

.list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: rem(0 0 $spacing-x-small);
  }
}

@include mq($from: xs) {
  .list {
    column-count: 2;
    column-gap: rem($spacing-medium);
  }
}

@include mq($from: md) {
  .wrapper {
    float: left;
    margin: 0;
    padding: rem(0 $spacing-large);
    width: 50%;
  }

  .list {
    column-count: 2;
    column-gap: rem($spacing-medium);
  }
}
