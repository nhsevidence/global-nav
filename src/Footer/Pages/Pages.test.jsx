import React from "react";

import Pages from "./Pages";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Pages", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<Pages />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot", () => {
		const wrapper = shallow(<Pages />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Uses overridden feedback URL as href for site with custom feedback URL", () => {
		const wrapper = shallow(<Pages service="cks" />);
		expect(
			wrapper.find("TrackedLink[children='Leave feedback']").prop("href")
		).toEqual("https://www.nice.org.uk/cks-feedback");
	});
});
