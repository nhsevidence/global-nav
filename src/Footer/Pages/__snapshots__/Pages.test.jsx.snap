// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Matches snapshot 1`] = `
<nav
  className="wrapper"
>
  <ul
    className="list"
  >
    <li
      key="About NICE"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="About NICE"
        href="https://www.nice.org.uk/about"
      >
        About NICE
      </TrackedLink>
    </li>
    <li
      key="Into practice"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Into practice"
        href="https://www.nice.org.uk/about/what-we-do/into-practice"
      >
        Into practice
      </TrackedLink>
    </li>
    <li
      key="Get involved with NICE"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Get involved with NICE"
        href="https://www.nice.org.uk/get-involved"
      >
        Get involved with NICE
      </TrackedLink>
    </li>
    <li
      key="Jobs"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Jobs"
        href="https://www.nice.org.uk/get-involved/jobs"
      >
        Jobs
      </TrackedLink>
    </li>
    <li
      key="News"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="News"
        href="https://www.nice.org.uk/news"
      >
        News
      </TrackedLink>
    </li>
    <li
      key="Newsletters and alerts"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Newsletters and alerts"
        href="https://www.nice.org.uk/news/nice-newsletters-and-alerts"
      >
        Newsletters and alerts
      </TrackedLink>
    </li>
    <li
      key="Contact us"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Contact us"
        href="https://www.nice.org.uk/get-involved/contact-us"
      >
        Contact us
      </TrackedLink>
    </li>
    <li
      key="Leave feedback"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Leave feedback"
        href="https://www.nice.org.uk/leave-feedback"
      >
        Leave feedback
      </TrackedLink>
    </li>
    <li
      key="Reusing our content"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Reusing our content"
        href="https://www.nice.org.uk/re-using-our-content"
      >
        Reusing our content
      </TrackedLink>
    </li>
    <li
      key="NICE UK Open Content License"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="NICE UK Open Content License"
        href="https://www.nice.org.uk/re-using-our-content/uk-open-content-licence"
      >
        NICE UK Open Content License
      </TrackedLink>
    </li>
    <li
      key="Syndicate our content"
    >
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Syndicate our content"
        href="https://www.nice.org.uk/about/what-we-do/nice-syndication-api"
      >
        Syndicate our content
      </TrackedLink>
    </li>
  </ul>
</nav>
`;
